applications/SwimmingDEMApplication/custom_elements/qsvms_dem_coupled.cpp

```diff
@@ -0,0 +1,859 @@
+
+        }
+        else {
+
+            if (rElementalDofList.size() != LocalSize)
+                rElementalDofList.resize(LocalSize);
+
+            unsigned int LocalIndex = 0;
+
+            for (unsigned int iNode = 0; iNode < NumNodes; ++iNode)
+            {
+                rElementalDofList[LocalIndex++] = this->GetGeometry()[iNode].pGetDof(VELOCITY_LAPLACIAN_X);
+                rElementalDofList[LocalIndex++] = this->GetGeometry()[iNode].pGetDof(VELOCITY_LAPLACIAN_Y);
+                if (Dim == 3) rElementalDofList[LocalIndex++] = this->GetGeometry()[iNode].pGetDof(VELOCITY_LAPLACIAN_Z);
+            }
+        }
+}
+
+template< class TElementData >
+std::string QSVMSDEMCoupled<TElementData>::Info() const
+{
+    std::stringstream buffer;
+    buffer << "QSVMSDEMCoupled #" << this->Id();
+    return buffer.str();
+}
+
+
+template< class TElementData >
+void QSVMSDEMCoupled<TElementData>::PrintInfo(std::ostream& rOStream) const
+{
+    rOStream << "QSVMSDEMCoupled" << Dim << "D";
+}
+
+/**
+ * Returns the squared element size, estimated as h^2 = 2*Area
+ * Returns the squared element size, estimated from the assumption V = (1/6) * h^3
+ * @see QSVMSDEMCoupled::FilterWidth
+ */
+template <class TElementData>
+double QSVMSDEMCoupled<TElementData>::FilterWidth()
+{
+    if (Dim == 2){
+        double FilterWidth = GeometryUtils::CalculateVolume2D(this->GetGeometry());
+        return 2.0 * FilterWidth;
+    } else {
+        const double TwoThirds = 2.0 / 3.0;
+        double FilterWidth = GeometryUtils::CalculateVolume3D(this->GetGeometry());
+        FilterWidth *= 6.0;
+        return pow(FilterWidth, TwoThirds);
+    }
+}
+/**
+ * Returns the square of the minimum element height, to be used as filter width in the Smagorinsky model
+ * @see QSVMSDEMCoupled::FilterWidth
+ */
+template <class TElementData>
+double QSVMSDEMCoupled<TElementData>::FilterWidth(const BoundedMatrix<double, NumNodes, Dim>& DN_DX)
+{
+    double inv_h_max = 0.0;
+    for(unsigned int i=0; i<NumNodes; i++)
+    {
+        double inv_h = 0.0;
+        for(unsigned int d=0; d<Dim; d++)
+            inv_h += DN_DX(i,d)*DN_DX(i,d);
+
+        if(inv_h > inv_h_max) inv_h_max = inv_h;
+    }
+
+    double DeltaSquared = 1.0/inv_h_max;
+
+    return DeltaSquared;
+}
+
+template <class TElementData>
+void QSVMSDEMCoupled<TElementData>::CalculateLocalSystem(MatrixType& rLeftHandSideMatrix,
+                                                        VectorType& rRightHandSideVector,
+                                                        ProcessInfo& rCurrentProcessInfo)
+    {
+        
+        QSVMS<TElementData>::CalculateLocalSystem(rLeftHandSideMatrix, rRightHandSideVector, rCurrentProcessInfo);
+        
+        if (rCurrentProcessInfo[FRACTIONAL_STEP] == 1) {
+            this->CalculateRightHandSide(rRightHandSideVector, rCurrentProcessInfo);
+        }else{
+            noalias(rLeftHandSideMatrix) = ZeroMatrix(LocalSize, LocalSize);
```

Comment:
Note that in this case you are calculating the standard QSVMS LHS for nothing. This is an overhead.